arb_os/output.mini

```diff
@@ -85,11 +86,14 @@ public impure func outputStats_endOfBlock(blockNum: uint, timestamp: uint) {
         updatedStats = updatedStats with {
             numLogs: updatedStats.numLogs + 1  // +1 for soon-to-be-issued block summary log
         };
+
+        emitBlockSummaryLog(blockNum, timestamp, globalBlockReceiptData.lastBlockNum);
+
         globalBlockReceiptData = globalBlockReceiptData with {
             statsAllTime: updatedStats
+        } with {
+            lastBlockNum: blockNum
```

Comment:
This reorders the call to `emitBlockSummaryLog` before the update to `globalBlockReceiptData`, but `emitBlockSummaryLog` depends on the value of that global so this effects the block logs. I think the original order should be restored

Reply:
Yes, that's a bug. Fixed it by splitting the update to globalBlockReceipt data into two pieces, one before emitting the block summary and one after.